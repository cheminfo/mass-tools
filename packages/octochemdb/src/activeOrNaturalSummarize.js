import { summarizeActivities } from './utils/summarizeActivities.js';
import { summarizePatents } from './utils/summarizePatents.js';
import { summarizePubMeds } from './utils/summarizePubMeds.js';
/**
 * @description This function summarizes the activities, patents and pubmeds of an entry of ActivesOrNaturals collection.
 * @param {object} entry - Entry to summarize
 * @param {string} term - Search term
 * @param {object} options - Options
 * @param {object} [options.activities] - Options for activities
 * @param {object} [options.patents] - Options for patents
 * @param {object} [options.pubmeds] - Options for pubmeds
 * @returns {Promise<Object>} - Summarized entry of ActivesOrNaturals collection
 */
export async function activeOrNaturalSummarize(entry, term, options = {}) {
  entry = { ...entry };

  let promises = [];

  promises.push(
    summarizeActivities(entry.data.activities, term, options.activities).then(
      (activities) => {
        entry.data.activities = activities;
      },
    ),
  );
  promises.push(
    summarizePatents(entry.data.patents, term, options.patents).then(
      (patents) => {
        entry.data.patents = patents;
      },
    ),
  );

  promises.push(
    summarizePubMeds(entry.data.pubmeds, term, options.pubmeds).then(
      (pubmeds) => {
        entry.data.pubmeds = pubmeds;
      },
    ),
  );

  await Promise.all(promises);
  return entry;
}
