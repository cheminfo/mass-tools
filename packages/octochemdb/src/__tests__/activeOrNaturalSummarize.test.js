import { readFileSync } from 'fs';
import { join } from 'path';

import { activeOrNaturalSummarize } from '../activeOrNaturalSummarize.js';

describe('activeOrNaturalSummarize', () => {
  it('term:anti', async () => {
    const entry = JSON.parse(
      readFileSync(join(__dirname, './details.json'), 'utf8'),
    );

    let options = {
      term: 'anti',
      fieldsPubMed: ['title', 'abstract', 'meshHeadings'],
      boostPubMed: {
        title: 2,
        abstract: 1,
        meshHeadings: 1,
      },
      fieldsPatent: ['title', 'abstract'],
      boostPatent: {
        title: 2,
        abstract: 1,
      },
      fieldsActivities: ['assay'],

      relevance: {
        k: 1.2,
        b: 0.75,
        d: 0.5,
      },
    };

    const result = await activeOrNaturalSummarize(entry, options);
    expect(result).toMatchSnapshot();
  });
  it('term:inhibition', async () => {
    const entry = JSON.parse(
      readFileSync(join(__dirname, './details.json'), 'utf8'),
    );

    let options = {
      term: 'inhibition',
      fieldsPubMed: ['title', 'abstract', 'meshHeadings'],
      boostPubMed: {
        title: 2,
        abstract: 1,
        meshHeadings: 1,
      },
      fieldsPatent: ['title', 'abstract'],
      boostPatent: {
        title: 2,
        abstract: 1,
      },
      fieldsActivities: ['assay'],

      relevance: {
        k: 1.2,
        b: 0.75,
        d: 0.5,
      },
    };

    const result = await activeOrNaturalSummarize(entry, options);
    expect(result).toMatchSnapshot();
  });
});
