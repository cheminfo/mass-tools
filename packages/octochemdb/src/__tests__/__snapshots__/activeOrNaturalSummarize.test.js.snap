// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`activeOrNaturalSummarize > term:anti 1`] = `
[
  {
    "document": {
      "$id": "CN-113226378-A",
      "abstract": "The present invention relates to antibody-drug conjugates comprising: (i) an antibody or antigen-binding fragment thereof, (ii) a polymer comprising specific repeat units, optionally covalently bound by a linker to one or more biologically active moieties, such as a small molecule drug, and (iii) a polymer-antibody linker moiety, covalently bound to both the polymer and the antibody or antigen-binding fragment thereof. Furthermore, the present invention relates to pharmaceutical compositions comprising the antibody-drug conjugates and to the use of the antibody-drug conjugates in medicine.",
      "title": "Antibody-drug conjugates",
      "url": "https://pubchem.ncbi.nlm.nih.gov/patent/CN-113226378-A",
    },
    "score": 11.673095920550296,
  },
  {
    "document": {
      "$id": "CN-107043371-A",
      "abstract": "The invention discloses the alkaloid shown in a kind of Formulas I or Formula II, its analog, isomers, raceme or derivative\\\\uFF1B Above-claimed cpd system the present invention it is isolated from purslane first, and study discovery above-claimed cpd there is \\\\u03B2 2 AR excitements and the difunctional activity of anti-inflammatory, the diseases associated with inflammation for the respiratory diseases such as treatment asthma and correlation provide useful compound.",
      "title": "\\\\u03B22AR excitements and the difunctional alkaloid of anti-inflammatory and its application",
      "url": "https://pubchem.ncbi.nlm.nih.gov/patent/CN-107043371-A",
    },
    "score": 7.132761847318311,
  },
  {
    "document": {
      "$id": "CN-107043371-B",
      "abstract": "The invention discloses alkaloid shown in formula I or formula II, and analogues, isomers, racemates or derivatives thereof;n         the compound is obtained by separating purslane for the first time, and the research finds that the compound has \\\\u03B2n         2 The double-function activities of AR excitement and anti-inflammation provide useful compounds for treating respiratory diseases such as asthma and related inflammatory diseases.",
      "title": "\\\\u03B2 2-AR agonism and anti-inflammation bifunctional alkaloid and application thereof",
      "url": "https://pubchem.ncbi.nlm.nih.gov/patent/CN-107043371-B",
    },
    "score": 6.425570096830262,
  },
  {
    "document": {
      "$id": "WO-2015198346-A1",
      "abstract": "The invention relates to herbal anti-adipogenic and pro-lipolytic composition(s) or phytochemical ingredients(s) comprising at least one component selected from the extract(s), fraction(s) and active compound(s) derived from  Alangium salvifolium  either alone or in combination with atleast one component selected from pharmaceutically or dietetically acceptable phytochemical actives, vehicle, diluent and carrier. The present novel ingredients or compositions can be used to control or treat overweight, obesity, metabolic syndrome or other metabolic disorders, regulate energy expenditure, prevention of atherosclerotic plaques in coronary artery and abdominal aorta, increase insulin sensitivity, improve glucose tolerance, lower triglyceride levels and balance glucose levels in mammals.",
      "title": "A composition comprising extract of alangium salvifolium having anti-adipogenic or anti-obesic activity",
      "url": "https://pubchem.ncbi.nlm.nih.gov/patent/WO-2015198346-A1",
    },
    "score": 5.260563620484841,
  },
  {
    "document": {
      "$id": "CN-108391756-A",
      "abstract": "The present invention provides a kind of milking sow protein source additive, and the raw material for preparing of the protein source additive includes at least\\\\uFF1AAmino acid and its derivative, carbohydrate, the aromatic rings of hydroxyl, nafoxidine acetic acid, alkaloid, 1 (4 methoxyphenyl) piperazine, phytosterol and 4 hydroxyl \\\\u03B2 damascones.Feed addictive provided by the invention can promote feed intake, improve efficiency of feed utilization, improve meat, while under the premise of being not required to that antibiotics is added, animal immunizing power and oxidation resistance can be significantly improved, reduction incidence, drug residue free, meat are safer high-quality.",
      "title": "A kind of preparation and application of milking sow protein source additive",
      "url": "https://pubchem.ncbi.nlm.nih.gov/patent/CN-108391756-A",
    },
    "score": 1.5979211411595906,
  },
  {
    "document": {
      "$id": "CN-102813929-B",
      "abstract": "The invention belongs to the technical field of medicine, relates to a low-concentration PEG-lipid derivative and an application thereof, particularly to a prescription composition and an application thereof which can relieving or avoiding accelerated blood clearance (ABC) of PEGylation preparations. According to the low-concentration PEG-lipid derivative and the application thereof, the low-content PEG-lipid derivative is used for modifying liquid particle preparations such as emulsions, liposome, vesicles, micro emulsions, micelles and nano-particles, PEG and lipid chain segments are connected through ester bonds, amido bonds or ether bonds in the PEG-lipid derivative, and the molecular weight of the PEG is 200-20000 Dalton. Preparations prepared by the PEG-lipid derivative can only cause slight or cannot cause the ABC, namely, can reduce or avoid the ABC after repeated injection. The low-concentration PEG does not reduce in vitro anti-tumor activity of preparations, and stability of the preparation can be improved if the low-concentration PEG-lipid derivative is added, particularly shaking instability of antagonistic emulsions.",
      "title": "Low-concentration polyethylene glycol-lipid (PEG-lipid) derivative and application thereof",
      "url": "https://pubchem.ncbi.nlm.nih.gov/patent/CN-102813929-B",
    },
    "score": 1.4575057650844445,
  },
  {
    "document": {
      "$id": "CN-112159699-A",
      "abstract": "The invention discloses a high-efficiency water-soluble magnetic shoe release agent and a preparation method thereof, wherein the preparation method comprises the following raw materials by weight: mineral oil: 50-70%; lubricant: 5-15%; film-forming emulsion: 5-10%; antirust agent: 1-4%; coupling agent: 1-3%; metal passivator: 0.1-1%; and (3) antiseptic bactericide: 0.5-3%; and (3) anti-foaming agent: 0.001-0.1%; pH stabilizer: 2-6%; nonionic surfactant: 2-6%; hard water resisting agent: 0.5-1%; anionic surfactant: 2-5%; auxiliary agent: 1 to 3 percent. Compared with the prior art, the release agent prepared by the invention has the advantages of high dilution ratio, low consumption, good release effect, complete magnetic shoe forming, bright surface, effectively improved yield, strong corrosion resistance, strong storage stability and good alkali storage performance; the detergent has the advantages of strong hard water resistance, strong cleaning capability, difficult oil and soap separation, stable emulsion, reduced machine halt time for cleaning a machine table, effectively improved production efficiency and reduced enterprise production cost.",
      "title": "Efficient water-soluble magnetic shoe release agent and preparation method thereof",
      "url": "https://pubchem.ncbi.nlm.nih.gov/patent/CN-112159699-A",
    },
    "score": 1.0587706838293096,
  },
  {
    "document": {
      "$id": "11008879",
      "journalInfo": {
        "Country": "Hungary",
        "ISSNLinking": "1216-8068",
        "MedlineTA": "Neurobiology (Bp)",
        "NlmUniqueID": 9312159,
      },
      "meshHeadings": [
        {
          "descriptorName": "Antioxidants",
        },
        {
          "descriptorName": "Antiparkinson Agents",
        },
        {
          "descriptorName": "Apoptosis",
        },
        {
          "descriptorName": "DNA Damage",
        },
        {
          "descriptorName": "DNA Fragmentation",
        },
        {
          "descriptorName": "Humans",
        },
        {
          "descriptorName": "Neurotoxins",
          "qualifierName": "pharmacology",
        },
        {
          "descriptorName": "Parkinson Disease",
          "qualifierName": "drug therapy",
        },
        {
          "descriptorName": "Salsoline Alkaloids",
          "qualifierName": "pharmacology",
        },
        {
          "descriptorName": "Selegiline",
        },
        {
          "descriptorName": "Structure-Activity Relationship",
        },
        {
          "descriptorName": "Tetrahydroisoquinolines",
        },
        {
          "descriptorName": "Tumor Cells, Cultured",
        },
      ],
      "nbCompounds": 12,
      "title": "Anti-apoptotic function of L-(-)deprenyl (Selegiline) and related compounds.",
      "url": " https://pubmed.ncbi.nlm.nih.gov/11008879",
    },
    "score": 1.4440208736340374,
  },
  {
    "document": {
      "$id": "18098162",
      "journalInfo": {
        "Country": "England",
        "ISSNLinking": "0749-1581",
        "MedlineTA": "Magn Reson Chem",
        "NlmUniqueID": 9882600,
      },
      "meshHeadings": [
        {
          "descriptorName": "Antioxidants",
        },
        {
          "descriptorName": "Deuterium",
        },
        {
          "descriptorName": "Magnetic Resonance Spectroscopy",
          "qualifierName": "methods",
        },
        {
          "descriptorName": "Molecular Structure",
        },
        {
          "descriptorName": "Oxidation-Reduction",
        },
        {
          "descriptorName": "Salsola",
          "qualifierName": "chemistry",
        },
        {
          "descriptorName": "Salsoline Alkaloids",
        },
        {
          "descriptorName": "Triterpenes",
        },
      ],
      "nbCompounds": 8,
      "title": "Structural determination of salsolins A and B, new antioxidant polyoxygenated triterpenes from Salsola baryosma, by 1D and 2D NMR spectroscopy.",
      "url": " https://pubmed.ncbi.nlm.nih.gov/18098162",
    },
    "score": 1.0764274563800005,
  },
  {
    "document": {
      "$id": "15843867",
      "journalInfo": {
        "Country": "Austria",
        "ISSNLinking": "0300-9564",
        "MedlineTA": "J Neural Transm (Vienna)",
        "NlmUniqueID": 9702341,
      },
      "meshHeadings": [
        {
          "descriptorName": "Apoptosis",
        },
        {
          "descriptorName": "Apoptosis Regulatory Proteins",
        },
        {
          "descriptorName": "Dose-Response Relationship, Drug",
        },
        {
          "descriptorName": "Humans",
        },
        {
          "descriptorName": "Mitochondria",
        },
        {
          "descriptorName": "Mitochondrial Membranes",
        },
        {
          "descriptorName": "Neuroprotective Agents",
          "qualifierName": "pharmacology",
        },
        {
          "descriptorName": "Pargyline",
        },
        {
          "descriptorName": "Propylamines",
          "qualifierName": "pharmacology",
        },
        {
          "descriptorName": "Proto-Oncogene Proteins c-bcl-2",
        },
        {
          "descriptorName": "Salsoline Alkaloids",
          "qualifierName": "toxicity",
        },
        {
          "descriptorName": "Tetrahydroisoquinolines",
          "qualifierName": "toxicity",
        },
        {
          "descriptorName": "Tumor Cells, Cultured",
        },
      ],
      "nbCompounds": 16,
      "title": "N-Propargylamine protects SH-SY5Y cells from apoptosis induced by an endogenous neurotoxin, N-methyl(R)salsolinol, through stabilization of mitochondrial membrane and induction of anti-apoptotic Bcl-2.",
      "url": " https://pubmed.ncbi.nlm.nih.gov/15843867",
    },
    "score": 0.7289677406517564,
  },
  {
    "document": {
      "$id": "11520893",
      "journalInfo": {
        "Country": "England",
        "ISSNLinking": "0022-3042",
        "MedlineTA": "J Neurochem",
        "NlmUniqueID": "2985190R",
      },
      "meshHeadings": [
        {
          "descriptorName": "Active Transport, Cell Nucleus",
          "qualifierName": "physiology",
        },
        {
          "descriptorName": "Animals",
        },
        {
          "descriptorName": "Apoptosis",
        },
        {
          "descriptorName": "Cell Line",
        },
        {
          "descriptorName": "Cell Nucleus",
          "qualifierName": "enzymology",
        },
        {
          "descriptorName": "Glyceraldehyde-3-Phosphate Dehydrogenases",
          "qualifierName": "metabolism",
        },
        {
          "descriptorName": "Humans",
        },
        {
          "descriptorName": "Immunoblotting",
        },
        {
          "descriptorName": "Immunohistochemistry",
        },
        {
          "descriptorName": "Indans",
        },
        {
          "descriptorName": "Membrane Potentials",
          "qualifierName": "drug effects",
        },
        {
          "descriptorName": "Mitochondria",
          "qualifierName": "metabolism",
        },
        {
          "descriptorName": "Neuroprotective Agents",
          "qualifierName": "pharmacology",
        },
        {
          "descriptorName": "Neurotoxins",
        },
        {
          "descriptorName": "Parkinson Disease",
          "qualifierName": "drug therapy",
        },
        {
          "descriptorName": "Proto-Oncogene Proteins c-bcl-2",
        },
        {
          "descriptorName": "Salsoline Alkaloids",
        },
        {
          "descriptorName": "Signal Transduction",
        },
        {
          "descriptorName": "Tetrahydroisoquinolines",
        },
        {
          "descriptorName": "Transfection",
        },
      ],
      "nbCompounds": 18,
      "title": "Transfection-enforced Bcl-2 overexpression and an anti-Parkinson drug, rasagiline, prevent nuclear accumulation of glyceraldehyde-3-phosphate dehydrogenase induced by an endogenous dopaminergic neurotoxin, N-methyl(R)salsolinol.",
      "url": " https://pubmed.ncbi.nlm.nih.gov/11520893",
    },
    "score": 0.666899903532947,
  },
]
`;

exports[`activeOrNaturalSummarize > term:inhibition 1`] = `
[
  {
    "document": {
      "$id": "NPC294249",
      "assay": "Inhibition : 10.75 %",
      "url": "http://bidd.group/NPASS/compound.php?compoundID=NPC294249",
    },
    "score": 1.172060050051001,
  },
  {
    "document": {
      "$id": "NPC294249",
      "assay": "Inhibition : -4.77 %",
      "url": "http://bidd.group/NPASS/compound.php?compoundID=NPC294249",
    },
    "score": 1.172060050051001,
  },
]
`;
